File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.java
class Solution {
    public long countBadPairs(int[] nums) {
        HashMap<Long,Integer> hs=new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            long val=i+1-nums[i];
            hs.put(val,hs.getOrDefault(val,0)+1);
        }
        
        long sum=0;
        for(long val:hs.values())
        {
            sum+=val*(val-1)/2;
        }
        long total=nums.length;
        total*=(nums.length-1);
        total/=2;
        return total-sum;
    }
}

File: 2364-count-number-of-bad-pairs/2364-count-number-of-bad-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
    var totalPair=(pair)=>{
        return pair*(pair-1)/2;
    }
var countBadPairs = function(nums) {
    
    const map = new Map();
    for(let i=0;i<nums.length;i++)
        {
            let val=i-nums[i];
            if(map.has(val))
                map.set(val,map.get(val)+1);
            else
                map.set(val,1);
        }
    
    let sum=totalPair(nums.length);
    
    for(val of map.values())
        sum-=totalPair(val);
    return sum;
};